python/README.md

```diff
@@ -1,73 +1,181 @@
 
 ## Configuration Options
 
+### Command Line Arguments
+
 ```
-usage: example.py [-h] [-host HOSTNAME] [-port PORT] -user USERNAME -pass PASSWORD
-                  -pat TOKEN -query QUERY [-tls]
-                  [-dcv DISABLE_CERTIFICATE_VERIFICATION]
-                  [-path_to_certs PATH_TO_CERTS] [-sp [SESSION_PROPERTIES ...]]
-                  [-engine ENGINE]
+usage: example.py [-config CONFIG_REL_PATH]
 
 options:
-  -h, --help            show this help message and exit
-  -host HOSTNAME, --hostname HOSTNAME
-                        Dremio co-ordinator hostname. Defaults to "localhost".
-  -port PORT, --flightport PORT
-                        Dremio flight server port. Defaults to 32010.
-  -user USERNAME, --username USERNAME
-                        Dremio username. Not applicable when connecting to Dremio
-                        Cloud
-  -pass PASSWORD, --password PASSWORD
-                        Dremio password. Not applicable when connecting to Dremio
-                        Cloud
-  -pat TOKEN, --token TOKEN
-                        Either a Personal Access Token or an OAuth2 Token. Only
-                        applicable to Dremio Cloud. Use --password if connecting to
-                        Dremio Software using PAT
-  -query QUERY, --sqlQuery QUERY
-                        SQL query to test. Must be enclosed in single quotes. If
-                        single quotes are already present within the query, change
-                        those to double quotes and enclose entire query in single
-                        quotes.
-  -tls, --tls           Enable encrypted connection. Defaults to False.
-  -dcv DISABLE_CERTIFICATE_VERIFICATION, --disableCertificateVerification DISABLE_CERTIFICATE_VERIFICATION
-                        Disables TLS server verification. Defaults to False.
-  -path_to_certs PATH_TO_CERTS, --trustedCertificates PATH_TO_CERTS
-                        Path to trusted certificates for encrypted connection.
-                        Defaults to system certificates.
-  -sp [SESSION_PROPERTIES ...], --sessionProperty [SESSION_PROPERTIES ...]
-                        Key value pairs of SessionProperty, example: -sp
-                        schema='Samples."samples.dremio.com"' -sp key=value
-  -engine ENGINE, --engine ENGINE
-                        The specific engine to run against. Only applicable to Dremio
-                        Cloud.
+  -config CONFIG_REL_PATH, --config-path CONFIG_REL_PATH            Set the relative path to the config file. Default is "./config.yaml".
+```
+
+### Config File Options
+
+**File Format**
+
+This example's config file uses the [YAML](https://yaml.org/) format.
+
+**Options**
+
+```
+hostname: <str>
+port: <int>
+username: <str>
+password: <str>
+token: <str>
+query: <str>
+tls: <bool>
+disable_certificate_verification: <bool>
+path_to_certs: <str>
+session_properties:
+  - <str>
+  - <str>
+engine: <str>
+```
```

Comment:
This code block seems redundant if we have the more detailed breakdown below.

Reply:
I think it gives a quick glance over all the properties, and lays out how the config.yaml file should look like.

Mongodb uses a [similar documentation style for their config file](https://www.mongodb.com/docs/manual/reference/configuration-options/#core-options).